### PR TITLE
Add `Host-/Servicegroup` grid views

### LINKED ISSUE #221: Host- and Servicegroup grid views missing
Screenshot from 2021-06-29 09-52-51 Screenshot from 2021-06-29 09-52-34

### PR DESCRIPTION
**Pull request by yhabteab**
resolves #221 

Blocked by: Icinga/ipl-web#154, Icinga/ipl-web#155

### CODE REVIEW

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: public/css/widget/group-grid.less [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          }
+
+          > .content a {
+            font-weight: normal;
+            line-height: 2;
+            display: inline-block;
+            max-width: 10em;
+            overflow: hidden;
+            white-space: nowrap;
+            text-overflow: ellipsis;
```
Comment (nilmerg) [inline public/css/widget/group-grid.less:41]: There's a mixin for this

File: public/css/widget/group-grid.less
Comment (nilmerg) [inline public/css/widget/group-grid.less:1]: I'd like to discuss the change requests here in person, so please notify me before working on them.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: library/Icingadb/Widget/ItemTable/ServicegroupGridCell.php
Comment (nilmerg) [inline library/Icingadb/Widget/ItemTable/ServicegroupGridCell.php:1]: Please use the `Translation` trait in here, instead of `t` and `tp`.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: library/Icingadb/Web/Control/ViewModeSwitcher.php [hunk 2]
```diff
@@ -139,7 +139,7 @@ public function setIdProtector(callable $protector): self
         return $this;
     }
 
-    private function protectId($id)
+    protected function protectId($id)
```
[the remark this answers is not in the document]
Comment (raviks789) [reply library/Icingadb/Web/Control/ViewModeSwitcher.php]: True

File: public/css/widget/table-layout.less [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    margin-top: -.5em;
+    margin-bottom: -.5em;
+
+    .title {
+      a:hover {
+        color: @icinga-blue;
+        text-decoration: none;
+      }
+    }
+  }
```
[the remark this answers is not in the document]
Comment (raviks789) [reply public/css/widget/table-layout.less]: I guess this was incorrectly introduced. This has been removed now.

**Review by nilmerg — CHANGES_REQUESTED** [review]
The table layout is broken now

**Review by nilmerg — APPROVED** [review]

**Pull request merged**